### PR TITLE
Add CI coverage for Windows

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Update CI workflow to run jobs on Windows as well

Note: Certain jobs from the CI workflow can only be run on Linx for now

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -32,9 +34,12 @@ jobs:
       run: rustup set profile minimal
     - uses: arduino/setup-protoc@v3
     - name: Test
-      run: ./scripts/test.sh
+      run: bash ./scripts/test.sh
   lint:
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ windows-latest, ubuntu-latest ]
```
Comment (cijothomas) [inline .github/workflows/ci.yml:41]: does it matter if lint checks are done on windows/linux/either?

Comment (utpilla) [reply .github/workflows/ci.yml]: Good point! Updated the workflow to run `docs` and `lint` jobs only on Linux.

**Review by lalitb — APPROVED** [review]
Thanks.

**Comment by cijothomas** [comment]
Consider doing windows for only the test/stable for now, given it is much slower. We can add it for all, later.
Also see what other repos like tracing-rs is doing.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -32,7 +34,7 @@ jobs:
       run: rustup set profile minimal
     - uses: arduino/setup-protoc@v3
     - name: Test
-      run: ./scripts/test.sh
+      run: bash ./scripts/test.sh
```
Comment (ThomsonTan) [inline .github/workflows/ci.yml:37]: Is there some scenario where the `bash` prefix is required?

Comment (utpilla) [reply .github/workflows/ci.yml:37]: `bash` prefix lets us run `.sh` files on Windows.

**Review by TommyCpp — APPROVED** [review]
Looks good. Thanks

**Comment by utpilla** [comment]
> Consider doing windows for only the test/stable for now, given it is much slower. We can add it for all, later. Also see what other repos like tracing-rs is doing.

@cijothomas I checked the tracing repo and for non-Linux platforms, they only test using the stable version of Rust. Check [this](https://github.com/tokio-rs/tracing/blob/master/.github/workflows/CI.yml#L194-L210). I have updated the workflow to do the same. I have also updated code coverage workflow to only run on Linux.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**